### PR TITLE
[WIP] Migrate to junit5

### PR DESCRIPTION
**Pull request by tokuhirom**

### CODE REVIEW

File: line-bot-api-client/src/test/java/com/linecorp/bot/client/CallbackAdaptorTest.java [hunk 1]
```diff
@@ -21,9 +21,9 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-import org.junit.Before;
 import org.junit.Rule;
```
Comment (imasahiro) [inline line-bot-api-client/src/test/java/com/linecorp/bot/client/CallbackAdaptorTest.java]: Q. Will you convert this as well in this PR?

File: line-bot-api-client/src/test/java/com/linecorp/bot/client/HeaderInterceptorWireMockTest.java [hunk 2]
```diff
@@ -23,7 +23,8 @@
 import java.net.URI;
 
 import org.junit.Rule;
```
Comment (imasahiro) [inline line-bot-api-client/src/test/java/com/linecorp/bot/client/HeaderInterceptorWireMockTest.java]: How about replace it with `@ExtendWith(MockitoExtension.class)`?

File: build.gradle [hunk 3]
```diff
@@ -108,7 +108,9 @@ subprojects {
         testImplementation 'com.github.stefanbirkner:system-rules'
         testImplementation 'com.squareup.okhttp3:mockwebserver'
         testImplementation 'org.hibernate.validator:hibernate-validator'
-        testImplementation 'org.springframework.boot:spring-boot-starter-test' // MockHttpServletRequest
+        testImplementation('org.springframework.boot:spring-boot-starter-test') { // MockHttpServletRequest
```
Comment (kazuki-ma) [inline build.gradle]: I think you can use `dependencyManagement` to define exclusion rule in this project.
This can remove copy&paste exclude rule.

**Comment by tokuhirom** [comment]
ref. https://github.com/square/okhttp/issues/4667

**Comment by imasahiro** [comment]
any update so far?

**Comment by tokuhirom** [comment]
okhttp team doesn't release new version since 2021/01 https://square.github.io/okhttp/changelog/

**Comment by tokuhirom** [comment]
Move to #692

**Pull request closed without merging**